plugins/outputs/prometheus_client/prometheus_client.go

```diff
@@ -242,86 +300,123 @@ func (p *PrometheusClient) Write(metrics []telegraf.Metric) error {
 			labels[sanitize(k)] = v
 		}
 
-		// Prometheus doesn't have a string value type, so convert string
-		// fields to labels.
-		for fn, fv := range point.Fields() {
-			switch fv := fv.(type) {
-			case string:
-				labels[sanitize(fn)] = fv
-			}
-		}
-
-		for fn, fv := range point.Fields() {
-			// Ignore string and bool fields.
-			var value float64
-			switch fv := fv.(type) {
-			case int64:
-				value = float64(fv)
-			case float64:
-				value = fv
-			default:
-				continue
+		switch point.Type() {
+		case telegraf.Summary:
+			var mname string
+			var sum float64
+			var count uint64
+			value := make(map[float64]float64)
+			for fn, fv := range point.Fields() {
+				switch fn {
+				case "sum":
+					sum = fv.(float64)
```

Comment:
We need to guard against this because we don't have full control that it will always be true, and in the other spots where we type assert.

Reply:
This one really got me.  Type assertions suck, and switching on .(type) seemed to be the only solution, but as you noted, we don't have control what the type would be which would result in a pretty long set of cases.  After rolling it around awhile (and not being a real in depth go guy), would doing something like `strconv.ParseFloat(fmt.Sprint(fv), 64)` be too silly or performance degrading?  (Convert the value to a string because you don't need to know the type for that, and then convert the string to a float64.  If it's not a number, presumably we've already converted it to a label.)

Reply:
I wouldn't do the string conversion, but you could do the two value return form: `f, ok := i.(float64)`. I guess we need to decide what we are going to do if a field is not a float (do we just drop it?).

You could also try switching over the type instead of the field name, this might end up reading better.

Reply:
The current functionality is to drop anything but an int64 or float64. (https://github.com/influxdata/telegraf/blob/master/plugins/outputs/prometheus_client/prometheus_client.go#L254)

That seems to have worked thus far, perhaps we just keep going with it until someone finds otherwise.  I've gone ahead and made the summary and histogram value handling the same as the gauge and counter handling.